diff --git a/index.js b/index.js
index 6202cab2438db3de7eeb173406c3d0e82662a0e5..d22b4e1e3be5d72166e6afb3fc18c73804a6e786 100644
--- a/index.js
+++ b/index.js
@@ -215,7 +215,7 @@ module.exports = function(gitPath) {
     committerDate: null,
     author: null,
     authorDate: null,
-    commitMessage: null,
+    commitMessage: '',
     root: null,
     commonGitDir: null,
     worktreeGitDir: null,
@@ -326,9 +326,9 @@ function getCommitData(gitPath, sha) {
             break;
           default:
             //should just be the commit message left
-            data.commitMessage = section;
+            data.commitMessage = data.commitMessage||'';
+            data.commitMessage += section+'\r\n';
         }
-
         return data;
       }, {});
   }